private/buf/bufcli/bufcli.go

```diff
@@ -607,11 +607,14 @@ func newRegistryProviderWithOptions(container appflag.Container, opts ...bufapic
 	return bufapiclient.NewConnectClientProvider(container.Logger(), client, options...)
 }
 
-// Returns a new provider function which, when invoked, returns an interceptor which will look up an auth token by
-// address and set it into the request header.  This is used for registry providers where the token is looked up by
-// the client address at the time of client construction (i.e. for clients where a user is already authenticated and
-// the token is stored in .netrc)
-func newAuthorizationInterceptorProvider(container appflag.Container) func(string) connect.UnaryInterceptorFunc {
+// NewAuthorizationInterceptorProvider returns a new provider function which, when invoked, returns an interceptor
+// which will look up an auth token by address and set it into the request header.  This is used for registry providers
+// where the token is looked up by the client address at the time of client construction (i.e. for clients where a
+// user is already authenticated and the token is stored in .netrc)
+//
+// Note that the interceptor returned from this provider is always applied LAST in the series of interceptors added to
+// a client.
+func NewAuthorizationInterceptorProvider(container appflag.Container) func(string) connect.UnaryInterceptorFunc {
```

Comment:
Thanks for adding a comment. 

Are these config option constructors used in other repos like core? Curious why make them exported..